### DISCUSSION TITLE
10.1.3 released - no high priority security patches

### DISCUSSION THREAD

**Discussion by jeredfloyd** [ianw1974/zimbra-build-scripts#103] [General]
10.1.3 is out.  Doesn't look like it requires an off-cycle build.  No word yet on fixing the iOS 18 mail performance issues, either.

A bit annoying that Zimbra doesn't pin FOSS repo package versions, so be careful when using your OS package manager for regular updates.  It's probably safe to pull in the upstream FOSS components out-of-sync but I notice there are updates to the spell components (again) which may require investigation.

**Comment by ianw1974** [comment]
What I've always found strange, is that a Zimbra installation cannot be performed without it actually downloading packages from Zimbra's repositories.  Supposedly the repositories were only meant to get updates for paid clients, since they stopped releasing patches through these.  So I kind of fail to see the reason why packages during an installation of a build made from source is pulling packages from the repositories.

You kind of wonder, why a build cannot be installed without relying on components from these repositories.  What is the build missing that it has to do this?  What is Zimbra not committing to the build repositories that we have to then download additional packages from deb/rpm repositories?  Are these additionally installed components open-source at this point?  If so, then why are they not included in the build package?

**Reply by jeredfloyd** [reply]
The zm-build-scripts are only for building the Zimbra-authored (or significantly modified) modules, like zmmailbox.

Zimbra also makes config changes and then ships a whole bunch of upstream FOSS projects, like Apache, nginx, clamav, etc. rather than using the OS-maintained version.  Some of this is because there are configuration changes, some of this is to avoid unexpected feature skew, some of this is because the file system paths and such are different.  This is a lot of overhead, but it's probably needed to support so many different Linux distros.... this way they can expect the same version of each FOSS component in any environment.  It does also mean Zimbra is responsible for tracking and pushing security updates for all of those FOSS components, and they will usually lag (often by months to years) the upstream.

When you do a zm-build, you are not building and packaging the FOSS components, only the Zimbra components.  The FOSS components are still fetched from Zimbra's FOSS repos.  Because this update behavior also affects paid Zimbra installs, I suspect there are pains taken to make sure the FOSS repos don't break older versions of the Zimbra components and it's safe to update with the OS package manager (although in general they would like users to update the Zimbra-owned components at the same time).  The most recent breakage we saw a few months ago is when the way the spell component was integrated changed in an incompatible way.

**Reply by ianw1974** [reply]
I remember using Zimbra pre rpm/deb repositories that had everything in it (probably Zimbra 7 or earlier), so it was doable in the past until someone introduced these repositories.

**Reply by adriangibanelbtactic** [reply]
> Are these additionally installed components open-source at this point?

Yes.

> What is Zimbra not committing to the build repositories that we have to then download additional packages from deb/rpm repositories?
> You kind of wonder, why a build cannot be installed without relying on components from these repositories.
> If so, then why are they not included in the build package?

## Some history

A long time ago the tgz installer for Zimbra was monolithic and included all of the additional packages. This implied that when a component (such as openssl) had to be updated you had to wait for Zimbra to build the big tgz installer, download it and install it.

Some of those packages were moved/migrated onto a repo so that you could just `apt-get update && apt-get dist-upgrade` it to be updated. Well, you still have that zimbra-patch (and similar packages in Zimbra NE repo) aberration.

## ZCS system is two parts

The ZCS system is basically two main parts that work together:
- The core packages
- The repo packages

For the FOSS part the core packages need to be built manually because Synacor no longer provides an official OSE binary.
The repo packages are built by Synacor and made public on a repo.

The repo packages can be built manually indeed but only a few companies like Zextras (back in the day), Beezim (in private) or Intalio build them themselves.

## How to build the repo

The Github repo regarding the repo packages can be found at: https://github.com/Zimbra/packages .
You can find my early attempt to automate this build at: https://github.com/maldua/zimbra-foss-packages-builder/tree/dev-v001 .

This failed for me, probably because it's not documented good enough because not so many people tries this build.
You can check in the Zimbra forum: https://forums.zimbra.org/viewtopic.php?t=72564 how far I got in my tests.

Eventually I might resume this project but not sure when.

**Reply by ianw1974** [reply]
I remember being asked about building those Zimbra/packages and you are right not documented enough, I prob got through about 3 package builds of the 50-70 packages or so, and also gave up simply because I didn't and don't have the time to spend on it.  Would have taken quite a bit of time to debug and figure out what was missing.

Wish you luck with that if you do try it again.

**Comment by ianw1974** [comment]
Spelling does seem to work though even after applying these updates.
